template/src/inngest/functions/users/sync-users.test.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * DISCLAIMER:
+ * The tests provided in this file are specifically designed for the `syncUsersPage` function example.
+ * These tests serve as a conceptual framework and are not intended to be used as definitive tests in a production environment.
+ * They are meant to illustrate potential test scenarios and methodologies that might be relevant for a SaaS integration.
+ * Developers should create their own tests tailored to the specific implementation details and requirements of their SaaS integration.
+ * The mock data, assertions, and scenarios used here are simplified and may not cover all edge cases or real-world complexities.
+ * It is crucial to expand upon these tests, adapting them to the actual logic and behaviors of your specific SaaS integration.
+ */
+import { expect, test, describe, vi } from 'vitest';
+import { createInngestFunctionMock } from '@elba-security/test-utils';
+import * as nangoAPI from '@/common/nango';
+import { syncUsers } from './sync-users';
+
+const organisationId = '00000000-0000-0000-0000-000000000001';
+const region = 'us';
+const nangoConnectionId = 'nango-connection-id';
+const syncStartedAt = Date.now();
+
+// TODO: Replace with your source-specific user type
+const users = Array.from({ length: 5 }, (_, i) => ({
+  id: `user-id-${i}`,
+  displayName: `User ${i}`,
+  email: `user-${i}@example.com`,
+}));
+
+const setup = createInngestFunctionMock(syncUsers, 'source/users.sync.requested');
+
+describe('syncUsers', () => {
+  test('should continue the sync when there is a next page', async () => {
+    // Mock Nango API to return valid credentials
+    // @ts-expect-error -- this is a mock
+    vi.spyOn(nangoAPI, 'nangoAPIClient', 'get').mockImplementation(() => ({
+      getConnection: vi.fn().mockResolvedValue({
+        credentials: { access_token: 'access-token' },
+      }),
+    }));
+
+    // TODO: Add your source-specific mocks here
+    // Example:
+    // vi.spyOn(usersConnector, 'getUsers').mockResolvedValue({
+    //   validUsers: users,
+    //   invalidUsers: [],
+    //   nextPage: 'next-page-token',
+    // });
+
```

Comment:
IMO, This code should be enabled since it is mandatory to  moc, this  getUsers function is common among the integrations, moc value can be differ
```suggestion
vi.spyOn(usersConnector, 'getUsers').mockResolvedValue({
      validUsers: users,
      invalidUsers: [],
      nextPage: null,
    });
```